src/test/java/jenkins/plugins/logstash/configuration/LogzioTest.java

```diff
@@ -0,0 +1,44 @@
+package jenkins.plugins.logstash.configuration;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class LogzioTest{
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private Logzio indexer;
+    private Logzio indexer2;
+
+    @Before
+    public void setup(){
+        indexer = new Logzio();
+        indexer.setHost("https://listener.logz.io:8071");
+        indexer.setKey("key");
+
+        indexer2 = new Logzio();
+        indexer2.setHost("https://listener.logz.io:8071");
+        indexer2.setKey("key");
+    }
+
+    @Test
+    public void sameSettingsAreEqual(){ assertThat(indexer.equals(indexer2), is(true)); }
+
+    @Test
+    public void keyChangeIsNotEqual() {
+        indexer.setKey("newPassword");
+        assertThat(indexer.equals(indexer2), is(false));
```

Comment:
` assertThat(indexer, is(not(indexer2)));` will print better failure messages than `expected true got false`

Reply:
The style I used is the same one used in ES, Rabbit, Redis, and Syslog

Reply:
there is a lot of legacy code out there, if you don't want to change it though I'm not insisting